keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1215 @@
+| Drop traffic             | Supported with a `Deny` rule action                                                                              | Supported via implicit isolation of target Pods                                    |
+| Skip enforcement         | Supported with an `Pass` rule action                                                                             | Not needed                                                                         |
+| Allow traffic            | Supported with an `Allow` rule action                                                                            | Default action for all rules is to allow                                           |
+| Implicit isolation       | No implicit isolation                                                                                            | All rules have an implicit isolation of target Pods                                |
+| Rule precedence          | Depends on the order in which they appear within a ANP                                                           | Rules are additive                                                                 |
+| Policy precedence        | Depends on `priority` field among ANPs. Enforced before K8s NetworkPolicies if positive numeric priority value   | Enforced after numeric-priority ClusterNetworkPolicies, before baseline-priority AdminNetworkPolicy |
+| Matching pod selection   | Can apply different rules to multiple groups of Pods                                                             | Applies rules to a single group of Pods                                            |
+| Rule identifiers         | Name per rule in string format. Unique within a ANP                                                              | Not supported                                                                      |
+| Cluster external traffic | Not supported                                                                                                    | Partially supported via IPBlock                                                    |
+| Namespace selectors      | Supports advanced selection of Namespaces with the use of `namespaceSet`                                         | Supports label based Namespace selection with the use of `namespaceSelector` field |
+
+Note that AdminNetworkPolicy can also apply to Pods in Namespaces that don't
+exist yet, and will automatically apply to a new Namespace as long as the new
+Namespace's labels match the AdminNetworkPolicy rule's appliedTo selection
+criteria. NetworkPolicies, on the contrary, only apply to Pods in the Namespace
+they are created in.
+
+### Notes/Constraints/Caveats
+
+It is important to note that the controller implementation for cluster-scoped
+policy APIs will not be provided as part of this KEP. Such controllers which
+realize the intent of these APIs will be provided by individual network policy
+providers, as is the case with the NetworkPolicy API.
```

Comment:
This is just a total distraction, but I wish there was a reference implementation, with no performance guarantees, to use for CI. But that's just me :-/

Reply:
+1 Maybe that's something we can actually put together :)

Reply:
We should be able top build a simulator which ingests the API and can be tested against

Reply:
... _history shows that ends being the reference implementation_ :)

Reply:
ahh, do you mean a controller only, without actually setting the rules?
that is a fantastic idea, I'm happy to participate on that

Reply:
@aojea I would like to actually write something like KPNG (maybe KPNP) as a reference upstream controller for network policy, maybe even in ebpf :)